src/uucore/src/lib/features/format/argument.rs

```diff
@@ -72,6 +75,19 @@ impl<'a, T: Iterator<Item = &'a FormatArgument>> ArgumentIter<'a> for T {
         }
     }
 
+    fn get_i64_non_negative(&mut self) -> i64 {
+        let Some(next) = self.next() else {
+            return 0;
+        };
+        match next {
+            FormatArgument::SignedInt(n) if !n.is_negative() => *n,
+            FormatArgument::Unparsed(s) if !s.starts_with('-') => {
```

Comment:
`!s.starts_with('-')` Can the string start with spaces? I think you should probably parse first...

Actually, you might be better off moving that logic to `resolve_asterisk_precision`, I feel it'd be easier to handle there.

Reply:
Oh I see, you even want to handle negative values `< i64::MIN`... Still need to figure out how to handle leading white spaces though (and please add a test for that ,-P)

Reply:
Interesting, I didn't realize GNU printf gave that error for significantly negative precision's - actually, it seems like I can remove get_i64_non_negative and still handle the case in the GNU tests with a precision of -2147483649.

Ignoring whitespace is unexpected for me, I don't think that's universal or documented in `man printf`? Either way, this is now tested.